File: prisma/refs/clone-graph.ts
type GraphNode = { val: number; neighbors: GraphNode[] };

export function cloneGraph(node: GraphNode | null): GraphNode | null {
  if (!node) return null;
  const map = new Map<GraphNode, GraphNode>();
  const dfs = (n: GraphNode): GraphNode => {
    const hit = map.get(n);
    if (hit) return hit;
    const copy: GraphNode = { val: n.val, neighbors: [] };
    map.set(n, copy);
    for (const nb of n.neighbors) copy.neighbors.push(dfs(nb));
    return copy;
  };
  return dfs(node);
}

File: prisma/refs/py/clone-graph.py
def clone_graph(node):
    if not node:
        return None
    mapping = {}

    def dfs(n):
        if id(n) in mapping:
            return mapping[id(n)]
        copy = Node(n.val)  # noqa: F821 (injected by harness)
        mapping[id(n)] = copy
        copy.neighbors = [dfs(nb) for nb in n.neighbors]
        return copy

    return dfs(node)
